shopinvader_payment_stripe/services/payment_stripe.py

```diff
@@ -0,0 +1,273 @@
+        """
+        Validator of confirm_payment service
+        target: see _allowed_payment_target()
+        payment_mode: The payment mode used to pay
+        stripe_payment_intent_id: The previously created intent
+        stripe_payment_method_id: The Stripe card created on client side
+        :return: dict
+        """
+        return {
+            "target": {
+                "type": "string",
+                "required": True,
+                "allowed": self._allowed_payment_target(),
+            },
+            "payment_mode": {"type": "string"},
+            "stripe_payment_intent_id": {"type": "string"},
+            "stripe_payment_method_id": {"type": "string"},
+        }
+
+    def _validator_return_confirm_payment(self):
+        return {
+            "requires_action": {"type": "boolean"},
+            "payment_intent_client_secret": {"type": "string"},
+            "success": {"type": "boolean"},
+            "error": {"type": "string"},
+        }
+
+    def _find_target(self):
+        """
+        Find the target object from a payment request dict
+        supported targets: current_cart
+        the returned object implements shopinvader.payable
+        :return: recordset
+        """
+        cart_component = self.component(usage="cart")
+        cart = cart_component._get()
+        return cart
+
+    def _get_formatted_amount(self, currency, amount):
+        """
+        The expected amount format by Stripe
+        :param amount: float
+        :return: int
+        """
+        res = int(
+            amount
+            if currency.name in INT_CURRENCIES
+            else float_round(amount * 100, 2)
+        )
+        return res
+
+    def _get_stripe_transaction_from_intent(self, intent):
+        """
+        Retrieve the transaction from intent string
+        :param intent: string
+        :return: payment.transaction
+        """
+        transaction = self.env["payment.transaction"].search(
+            [
+                ("acquirer_reference", "=", intent),
+                ("acquirer_id.provider", "=", "stripe"),
+            ],
+            limit=1,
+        )
+        return transaction
+
+    def _get_stripe_private_key(self, tx_data=False, tx=False):
+        """
+        Return stripe private key depending on tx dict (before creation) or
+        with payment.transaction recordset
+        :param tx_data: dict
+        :param tx: account.payment.mode
+        :return: string
+        """
+        acquirer = self.env["payment.acquirer"]
+        if tx:
+            acquirer = tx.acquirer_id
+        if tx_data and "payment_mode_id" in tx_data:
+            acquirer = acquirer.browse(tx_data.get("acquirer_id"))
+        return acquirer.filtered(
+            lambda a: a.provider == "stripe"
+        ).stripe_secret_key
+
+    def confirm_payment(
+        self,
+        target,
+        payment_mode=False,
+        stripe_payment_method_id=False,
+        stripe_payment_intent_id=False,
```

Comment:
@rousseldenis IMO confirm_payment should takes **kwargs as params since the schema is defined by cerberus. In this way, it's possible to use the same implementation for others kind of target that could requires additional parameters.